File: Easy/Array/832. Flipping an Image/Solution.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        
        for(int[] row : image){
            int n = row.length; 

            for(int i = 0; i < (n + 1) / 2; i++){
                int j = n - 1 - i;

                int temp = row[i];
                row[i] = row[j] ^ 1;
                row[j] = temp ^ 1;
            }
        }

        return image;
    }
}

File: Easy/Array/832. Flipping an Image/Solution.kt
class Solution {
    fun flipAndInvertImage(image: Array<IntArray>): Array<IntArray> {
        
        for(row in image){
            val n = row.size

            for(i in 0 until (n + 1) / 2){
                val j = n - 1 - i

                if(i == j){
                    row[i] = row[i] xor 1
                } else {
                    val temp = row[i]
                    row[i] = row[j] xor 1
                    row[j] = temp xor 1
                }
            }
        }

        return image
    }
}
